app/javascript/components/admin/AdminTable.js

```diff
@@ -0,0 +1,694 @@
+    const options = {
+      title: 'Reset Password(s)',
+      okLabel: 'Confirm',
+      cancelLabel: 'Cancel',
+    };
+    if (await confirmDialog(confirmText, options)) {
+      this.handleResetPasswords();
+    }
+  };
+
+  /**
+   * Called when the reset 2FA action is clicked.
+   * Opens a confirmation modal and calls method to reset 2FA for selected users if confirmed.
+   */
+  handleReset2FAClick = async () => {
+    const confirmText = `Are you sure you would like to reset two-factor authentication for ${this.state.table.selectedRows.length} user(s)?`;
+    const options = {
+      title: 'Reset Two-Factor Authentication',
+      okLabel: 'Confirm',
+      cancelLabel: 'Cancel',
+    };
+    if (await confirmDialog(confirmText, options)) {
+      this.handleReset2FA();
+    }
+  };
+
+  /**
+   * Called when the user confirms the password reset action.
+   * Makes a POST request to reset the password of all selected users.
+   */
+  handleResetPasswords = () => {
+    const path = 'reset_password';
+    const ids = this.state.table.selectedRows.map(row => {
+      return this.state.table.rowData[parseInt(row)].id;
+    });
+    const dataToSend = {
+      ids: ids,
+    };
+
+    const handleSuccess = () => {
+      toast.success(`Password reset for ${this.state.selectedRows} users.`, {
+        autoClose: 2000,
+        position: toast.POSITION.TOP_CENTER,
+      });
+    };
+
+    const handleError = error => {
+      toast.error(`Failed to reset password for ${this.state.selectedRows} users.`, {
+        autoClose: 2000,
+        position: toast.POSITION.TOP_CENTER,
+      });
+      console.log(error);
+    };
+
+    this.axiosAdminPostRequest(path, dataToSend, handleSuccess, handleError);
+  };
+
+  /**
+   * Called when the user confirms the 2FA reset action.
+   * Makes a POST request to reset 2FA for all selected users.
+   */
+  handleReset2FA = () => {
+    const path = 'reset_2fa';
+    const ids = this.state.table.selectedRows.map(row => {
+      return this.state.table.rowData[parseInt(row)].id;
+    });
+    const dataToSend = {
+      ids: ids,
+    };
+
+    const handleSuccess = () => {
+      toast.success(`Two-factor authentication reset for ${this.state.selectedRows} users.`, {
+        autoClose: 2000,
+        position: toast.POSITION.TOP_CENTER,
+      });
+    };
+
+    const handleError = error => {
+      toast.error(`Failed to reset Two-factor Authentication for ${this.state.selectedRows} users.`, {
+        autoClose: 2000,
+        position: toast.POSITION.TOP_CENTER,
+      });
+      console.log(error);
+    };
+
+    this.axiosAdminPostRequest(path, dataToSend, handleSuccess, handleError);
+  };
+
+  render() {
+    return (
+      <div>
+        <div className="d-flex justify-content-between mb-2">
+          <div className="mb-1">
+            <Button className="mr-1" size="lg" onClick={this.handleAddUserClick}>
```

Comment:
minor UI comment, but these buttons on the left are large while the search bar and actions button on the right are not. was thinking it might look better if these buttons have the same height?
![Screen Shot 2020-08-05 at 11 04 57 AM](https://user-images.githubusercontent.com/9956561/89429509-a1f4ab80-d70b-11ea-8e75-4bafab0488a1.png)

Reply:
Agreed, made the buttons the same size (medium) as search bar and actions button. Looks a lot better.